File: examples-sdk/python/101b-multi-pipeline-benchmark.py
#!/usr/bin/env python3
"""101b-multi-pipeline-benchmark — Python SDK equivalent
Compile: vil compile --from python --input 101b-multi-pipeline-benchmark.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

pipeline = VilPipeline("MultiPipelineBench", 3090)
pipeline.sink(port=3090, path="/trigger", name="gateway")
pipeline.source(url="http://127.0.0.1:4545/v1/chat/completions", json_tap="choices[0].delta.content", name="l_l_m_upstream")
pipeline.route("sink.trigger_out", "source.trigger_in", "LoanWrite")
pipeline.route("source.response_data_out", "sink.response_data_in", "LoanWrite")
pipeline.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")
pipeline.compile()

File: examples-sdk/typescript/101b-multi-pipeline-benchmark.ts
#!/usr/bin/env tsx
// 101b-multi-pipeline-benchmark — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 101b-multi-pipeline-benchmark.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const p = new VilPipeline("MultiPipelineBench", 3090);
p.sink({ port: 3090, path: "/trigger", name: "gateway" });
p.source({ name: "l_l_m_upstream", url: "http://127.0.0.1:4545/v1/chat/completions", jsonTap: "choices[0].delta.content" });
p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
p.compile();
